draft1 of the dtos

jbhatt-salesforce opened the pull request:
@saaja-sfdc please take a look. Is this the correct direction?

jbhatt-salesforce committed:
draft1 of the dtos

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Base.java:
```diff
@@ -0,0 +1,56 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+	 
+/*
```
Remove this placeholder javadoc and update the author tag.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,165 @@
+	    private String cronEntry;
+	    private List<BigInteger> suspensionLevelIdList = new ArrayList<BigInteger>();
+
+		public String getServcie() {
```
Spelling error in method name.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,165 @@
+ * @author  Ruofan Zhang (rzhang@salesforce.com)
+ */
+@SuppressWarnings("serial")
+//@JsonIgnoreProperties(ignoreUnknown = true)
```
You'll need to include faster jackson json as a dependency for this to resolve.  It shouldn't be commented out.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,165 @@
+import java.util.List;
 
+
+import com.salesforce.dva.warden.entity.SuspensionLevel;
```
This import should be removed.  The class doesn't exist  in the API's project.  It will cause the build to fail.

saaja-sfdc commented:
@jbhatt-salesforce This looks pretty good.  Please see my comments for the few minor issues I found.

jbhatt-salesforce committed:
Another draft

jbhatt-salesforce commented:
@saaja-sfdc adding some more changes. I added Infraction, SuspensionLevel. Modified Policy. I still have a problem with jalopy that i'm trying to figure out. and the other problem is with integration tests.

Commit:
Merge branch 'master' into addingdtos-draft1

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Entity.java:
```diff
@@ -0,0 +1,147 @@
+package com.salesforce.dva.warden.dto;
```
Header comment needs to start on line 1 in all files.  Anything like this package declaration has to occur after the header.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Entity.java:
```diff
@@ -0,0 +1,147 @@
+/**
+ * The base entity DTO.
+ *
+ * @author  Tom Valine (tvaline@salesforce.com)
```
Update the author to your name.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Infraction.java:
```diff
@@ -0,0 +1,57 @@
+ */
+
+@SuppressWarnings("serial")
+public class Infraction extends com.salesforce.dva.warden.dto.Entity{
```
You can use the simple name for entity here.  No need for fully qualified name.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Infraction.java:
```diff
@@ -0,0 +1,57 @@
+@SuppressWarnings("serial")
+public class Infraction extends com.salesforce.dva.warden.dto.Entity{
+	
+	BigInteger policy_id;
```
These should all use the 'private' access modifier, not package private.
Rename to policyId

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Infraction.java:
```diff
@@ -0,0 +1,57 @@
+	BigInteger infraction_timestamp;
+	BigInteger expiration_timestamp;
+	
+	public BigInteger getPolicy_id() {
```
Missing javadoc for all these methods.
rename to getPolicyId

saaja-sfdc replied:
method names should all be camel case.  e.g. getPolicyId()

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Infraction.java:
```diff
@@ -0,0 +1,57 @@
+	
+	BigInteger policy_id;
+	int user_id;
+	BigInteger infraction_timestamp;
```
Timestamps should be of type Long.
rename infractionTimestamp

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Infraction.java:
```diff
@@ -0,0 +1,57 @@
+public class Infraction extends com.salesforce.dva.warden.dto.Entity{
+	
+	BigInteger policy_id;
+	int user_id;
```
BigInteger userId;

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Infraction.java:
```diff
@@ -0,0 +1,57 @@
+	public void setUser_id(int user_id) {
+		this.user_id = user_id;
+	}
+	public BigInteger getInfraction_timestamp() {
```
timestamp getter/setter should use data type of Long

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
 package com.salesforce.dva.warden.dto;
```
Missing header

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+ */
+@SuppressWarnings("serial")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Policy extends com.salesforce.dva.warden.dto.Entity {
```
No need for fully qualified class name.  Can just use simple name 'Entity'.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+public class Policy extends com.salesforce.dva.warden.dto.Entity {
+	//~ Instance fields ******************************************************************************************************************************
+
+	    private String servcie;		
```
spelling error.  should be 'service'.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+
+	    private String servcie;		
+	    private String name;
+	    private List<String> owner  = new ArrayList<String>();
```
Collections should use plural form of name:  'owners' & 'users'

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+	    private String cronEntry;
+	    private List<BigInteger> suspensionLevels = new ArrayList<BigInteger>();
+
+		public String getServcie() {
```
rename to getService()

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+	    private String timeUnit;
+	    private double defaultValue;
+	    private String cronEntry;
+	    private List<BigInteger> suspensionLevels = new ArrayList<BigInteger>();
```
It looks like the other DTO's actually do use the 'id' naming convention.  Apologies, but please change this back to 'suspensionLevelIds'

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+			return servcie;
+		}
+
+		public void setService(String service) {
```
Add javadoc for all these methods.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+			return defaultValue;
+		}
+
+		public void setDefaultValue(double defaultValue) {
```
All number variables should use the autobox versions.  In this case, change type to Double.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+			return triggerType;
+		}
+
+		public void setTriggerType(String triggerType) {
```
Fields that correspond to enumerations, should restrict the values to the valid enumerated values.  For triggerType, the data type should be an enum that's added to the DTO.

public enum TriggerType {
  EQUAL,
  NOT_EQUAL,
  ...
}

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+			this.triggerType = triggerType;
+		}
+
+		public String getAggregator() {
```
Same for aggregator.  Probably should be an enum.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/Policy.java:
```diff
@@ -1,5 +1,164 @@
+			this.timeUnit = timeUnit;
+		}
+
+		public double getDefaultValue() {
```
type should be Double

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/SuspensionLevel.java:
```diff
@@ -0,0 +1,74 @@
+import java.math.BigInteger;
+
+@SuppressWarnings("serial")
+public class SuspensionLevel extends com.salesforce.dva.warden.dto.Entity{
```
No need for fully qualified class name.

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/SuspensionLevel.java:
```diff
@@ -0,0 +1,74 @@
+public class SuspensionLevel extends com.salesforce.dva.warden.dto.Entity{
+
+	BigInteger policy_id;
+	int level_number;
```
Use autobox type of Integer

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/SuspensionLevel.java:
```diff
@@ -0,0 +1,74 @@
+	int infraction_count;
+	BigInteger suspension_time;
+	
+	public BigInteger getPolicy_id() {
```
Use camelCase: getPolicyId()

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/SuspensionLevel.java:
```diff
@@ -0,0 +1,74 @@
+@SuppressWarnings("serial")
+public class SuspensionLevel extends com.salesforce.dva.warden.dto.Entity{
+
+	BigInteger policy_id;
```
Use camelCase 'policyId'

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/dto/SuspensionLevel.java:
```diff
@@ -0,0 +1,74 @@
+
+
+
+	public void setLevel_number(int level_number) {
```
type should be Integer

saaja-sfdc commented on WardenApi/src/main/java/com/salesforce/dva/warden/entity/Entity.java:
```diff
@@ -0,0 +1,5 @@
+package com.salesforce.dva.warden.entity;
```
Entities should not be in the API.  Delete com.salesforce.dva.warden.entity package and all its contents.

saaja-sfdc commented:
@jbhatt-salesforce I've added my comments.
